### PR TITLE
Bugfix/sanatize session id to filename

### PR DESCRIPTION
**Pull request by veryrusty**
Encode session ids before concatenating them into file paths for file-based session engines.

### CODE REVIEW

File: lib/Dancer2/FileUtils.pm [hunk 1]
```diff
@@ -87,6 +87,16 @@ sub normalize_path {
     return $path;
 }
 
+sub escape_filename {
+    my $filename = shift or return;
+
+    # based on escaping used in CHI::Driver. Our use-case is one-way,
+    # so we allow utf8 chars to be escaped, but NEVER do the inverse
+    # operation.
+    $filename =~ s/([^A-Za-z0-9_\=\-\~])/sprintf("+%02x", ord($1))/ge;
```
Comment (yanick) [inline lib/Dancer2/FileUtils.pm]: The regex can be simplified a wee bit:

```
$filename =~ s/([^a-z0-9_=~-])/ sprintf "+%02x", ord $1 /ige;
```

**Comment by veryrusty** [comment]
Thanks @yanick! Simpler is good :)

.. and I got the wrong Andrew in the commit message. :(

**Comment by xsawyerx** [comment]
:+1: I will merge and release as soon as I can.
(If not tonight, tomorrow.)

**Comment by xsawyerx** [comment]
Merged!

**Pull request merged**